MichaelKintscher/qualitative-coding-app
# Adds Dynamic Button Creation Functionality

This patch allows a user to create buttons with data based on the title of each column in the table. When a button is clicked, the first column of the table will always populate with the current time-stamp of the video being played, then each subsequent column will populate based on data assigned to a button.

Testing Steps:

  1. Run the application and clear all sessions
  2. Created a new session and name it "test"
  3. Click "File" -> "Load video file" then add a video to the application
  4. Click "+" in the "Coding Assistance Buttons" panel
  5. Name this button "button1", assign the hotkey "a",  leave the "Time" field blank and populate each field with data of your choice then click "Create". Do not close the dialog
  6. Repeat step 5 to create a second button with a different name, hotkey, and data. Close the dialog
  7. Press "a" on your keyboard and ensure the table populates with button1's data.
  8. Repeat step 7 with the second button's hotkey. Ensure the data populates and is different from button1's
  9. Click "-" in the "Coding Assistance Buttons" panel
  10. Delete "button1"
  11. Press "a" and ensure no data populates in the table
  12. Close the application

Type: New Functionality
